### PR TITLE
feat(@refinedev/cli): add "--platform" option to specify project type

### PR DESCRIPTION
**Pull request by mikeyfarina**
Adds CLI option to specify platform to run refine on

### Closing issues

closes https://github.com/refinedev/refine/issues/4795

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [ ] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [ ] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

**Comment by MahirMahdi** [comment]
> **IMPORTANT: Please do not create a Pull Request without creating an issue first.**
> 
> _Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._
> 
> Please provide enough information so that others can review your pull request:
> 
> Explain the **details** for making this change. What existing problem does the pull request solve?
> 
> ### Test plan (required)
> Demonstrate the code is solid. If not, please add `WIP:` in its title.
> 
> ### Closing issues
> closes #4795 
> 
> ### Self Check before Merge
> Please check all items below before review.
> 
> * [ ]  Corresponding issues are created/updated or not needed
> * [ ]  Docs are updated/provided or not needed
> * [ ]  Examples are updated/provided or not needed
> * [ ]  TypeScript definitions are updated/provided or not needed
> * [ ]  Tests are updated/provided or not needed
> * [ ]  Changesets are provided or not needed

**Comment by MahirMahdi** [comment]
closes #4795

File: .changeset/shiny-cooks-push.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/cli": minor
+---
+
+add "--platform" option to run, dev, and build to specify which project type to use for the commands
```
Comment (MahirMahdi) [inline .changeset/shiny-cooks-push.md:5]:
```suggestion
add "--platform" option to start, dev, and build to specify project type externally
```

**Review by MahirMahdi — CHANGES_REQUESTED** [review]
Looks good to me.

**Comment by mikeyfarina** [comment]
@MahirMahdi Thank you for the review!

**Comment by aliemir** [comment]
Hey @mikeyfarina thank you for the PR! 🚀  I'll review & test it soon. @MahirMahdi thank you for the review ❤️

**Comment by MahirMahdi** [comment]
@aliemir anytime ❤️

**Comment by mikeyfarina** [comment]
@aliemir @MahirMahdi Anything I can do to help get this into a place where I can utilize it in my project?

**Comment by mikeyfarina** [comment]
@aliemir @MahirMahdi Hi guys 😁 anything i can do to help get this pr into a state that I can use it in my other repo?

**Review by omeraplak — APPROVED** [review]

**Comment by MahirMahdi** [comment]
@mikeyfarina I guess you need to wait until it gets merged.

**Review by aliemir — APPROVED** [review]

**Pull request merged**